### PR TITLE
AFHS-429 - added children dob list in eligibility override

### PR DESCRIPTION
**Pull request by AmarkumarJagtap**

### CODE REVIEW

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/constraint/EligibilityOverrideValidatorTest.java [hunk 1]
```diff
@@ -47,14 +48,22 @@ void shouldFailToValidateEligibilityOverrideWithNullValues(EligibilityOverride e
                 EligibilityOverride.builder()
                         .eligibilityOutcome(null)
                         .overrideUntil(OVERRIDE_UNTIL_FIVE_YEARS)
+                        .childrenDob(emptyList())
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/entity/constraint/EligibilityOverrideValidatorTest.java]: I think this would be clearer with specific dates like `MAGGIE_AND_LISA_DOBS` rather than `emptyList()`, same below too.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/NewClaimDTOTest.java [hunk 2]
```diff
@@ -111,13 +130,16 @@ void shouldFailToValidateClaimWithEligibilityOverrideUntilDatesNotInFuture(Local
     @Test
     void shouldSuccessfullyValidateClaimWithEligibilityOverride() {
         //Given
-        NewClaimDTO claim = aValidClaimDTOWithEligibilityOverride(
-                EXPECTED_DELIVERY_DATE_IN_TWO_MONTHS,
-                NO_CHILDREN,
-                EligibilityOutcome.CONFIRMED,
-                OVERRIDE_UNTIL_FIVE_YEARS);
+        EligibilityOverrideDTO eligibilityOverrideDTO = EligibilityOverrideDTO.builder()
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/model/NewClaimDTOTest.java]: Can we have this in a test data factory please?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementServiceTest.java [hunk 3]
```diff
@@ -174,10 +179,12 @@ void shouldEvaluateClaimForGivenPaymentCycle() {
         verify(eligibilityAndEntitlementDecisionFactory).buildDecision(IDENTITY_AND_ELIGIBILITY_RESPONSE, VOUCHER_ENTITLEMENT, false);
     }
 
-    @Test
-    void shouldEvaluateClaimForGivenPaymentCycleWithEligibilityOverride() {
+    @ParameterizedTest
+    @MethodSource("childrenDobs")
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementServiceTest.java]: We expect no children to behave the same as having children looking at this test, just wondering why the need to parameterise this test?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java [hunk 4]
```diff
@@ -4,27 +4,39 @@
...
 
 import java.time.LocalDate;
+import java.util.List;
 
+import static java.util.Collections.emptyList;
 import static uk.gov.dhsc.htbhf.claimant.testsupport.TestConstants.OVERRIDE_UNTIL_FIVE_YEARS;
 
 public class EligibilityOverrideTestDataFactory {
 
     public static EligibilityOverride aConfirmedEligibilityOverride() {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java]: How about `aConfirmedEligibilityOverrideWithNoChildren`?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java [hunk 5]
```diff
@@ -4,27 +4,39 @@
...
+    }
+
+    public static EligibilityOverride aConfirmedEligibilityOverrideWithChildren(List<LocalDate> childrenDob) {
+        return aConfirmedEligibilityOverrideBuilder()
+                .overrideUntil(OVERRIDE_UNTIL_FIVE_YEARS)
+                .childrenDob(childrenDob)
                 .build();
     }
 
     public static EligibilityOverride aConfirmedEligibilityWithUntilDate(LocalDate overrideUntil) {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java]: Would it be worthwhile mentioning no children in this and the next method too?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestConstants.java [hunk 6]
```diff
@@ -26,5 +29,6 @@
     public static final String TEST_EXCEPTION_MESSAGE = "test exception";
     public static final RuntimeException TEST_EXCEPTION = new RuntimeException(TEST_EXCEPTION_MESSAGE);
     public static final LocalDate OVERRIDE_UNTIL_FIVE_YEARS = LocalDate.now().plusYears(5);
+    public static final List<LocalDate> CHILD_BORN_IN_FUTURE = singletonList(LocalDate.now().plusDays(1));
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestConstants.java]: Could we call this `CHILD_DOB_IN_FUTURE` please as being born is the future is somewhat impossible. :-)

File: swagger.yml [hunk 7]
```diff
@@ -378,6 +378,12 @@ definitions:
   EligibilityOverrideDTO:
     type: "object"
     properties:
+      childrenDob:
+        type: "array"
+        description: "The dates of birth of the claimant's children (if they have\
```
Comment (dwybourn) [inline swagger.yml]: Can we update the description to state that an empty list must be used if there are no children.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestConstants.java (hunk 6 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/TestConstants.java]: It's only used to validate that the future DOB is rejected - so I think it being impossible is valid :)

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementServiceTest.java (hunk 3 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/EligibilityAndEntitlementServiceTest.java]: we pass the list of childrenDobs into a constructor _and an assertion_ - so each test does assert that different children's dobs are used. Without adding this the original test would have passed if the code had hard-coded an empty list. I guess we could simply use a non-empty list, but the parameterised test doesn't hurt.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java (hunk 5 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/testsupport/EligibilityOverrideTestDataFactory.java]: Not for `aNotConfirmedEligibilityOverride` - there should never be children if eligibility is not confirmed.
